logrotate/src/LogRotateHelper.ipp

```diff
@@ -168,20 +164,22 @@ std::string LogRotateHelper::changeLogFile(const std::string& directory, const s
       file_name = log_prefix_backup_;
    }
 
-   std::string prospect_log = createLogFileName(directory + file_name);
```

Comment:
The only real change in this function is adding the 'createPath', previously called `pathSanityFix` (_the old name was bad_).  Now having the constructor also calling this function removes some duplication.